### PR TITLE
Assignment4 KS-D & 7hawks

### PR DESCRIPTION
**Pull request by KS-D**
collaborated with @7hawks

### CODE REVIEW

**Comment by KS-D** [comment]
We will review this.

File: SecretSanta/src/SecretSanta.Api/Startup.cs [hunk 1]
```diff
@@ -12,9 +19,27 @@ public class Startup
...
+            var sqliteConnection = new SqliteConnection("DataSource=:memory:");
+            sqliteConnection.Open();
+
+            services.AddDbContext<ApplicationDbContext>(options =>
+                options.EnableSensitiveDataLogging()
+                    .UseSqlite(sqliteConnection));
+
+            services.AddScoped<IGiftService, GiftService>();
+            services.AddScoped<IUserService, UserService>();
+            services.AddScoped<IGroupService, GroupService>();
```
Comment (KS-D) [inline SecretSanta/src/SecretSanta.Api/Startup.cs]: Services are registered with the proper scope.

Comment (KS-D) [inline SecretSanta/src/SecretSanta.Api/Startup.cs]: Services are added with the proper scoping and the database is properly registered.

File: SecretSanta/src/SecretSanta.Api/Startup.cs [hunk 2]
```diff
@@ -12,9 +19,27 @@ public class Startup
...
+
+            System.Type profileType = typeof(AutomapperConfigurationProfile);
+            System.Reflection.Assembly assembly = profileType.Assembly;
+            services.AddAutoMapper(new[] { assembly });
```
Comment (KS-D) [inline SecretSanta/src/SecretSanta.Api/Startup.cs]: Automapper properly registered.

File: SecretSanta/src/SecretSanta.Api/Startup.cs [hunk 3]
```diff
@@ -26,13 +51,11 @@ public static void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
-            app.UseEndpoints(endpoints =>
-            {
-                _ = endpoints.MapGet("/", async context =>
-                  {
-                      await context.Response.WriteAsync("Hello from API!");
-                  });
-            });
+            app.UseOpenApi();
+
+            app.UseSwaggerUi3();
```
Comment (KS-D) [inline SecretSanta/src/SecretSanta.Api/Startup.cs]: NSwag is set up in the services and called during the configure.

**Comment by 7hawks** [comment]
I will review this too.

File: SecretSanta/test/SecretSanta.Api.Tests/Controller/GiftControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    [TestClass]
+    public class GiftControllerTests
+    {
+
+        [TestMethod]
+        public void Create_Controller_Success()
+        {
+            var controller = new GiftController(new Mock<IGiftService>().Object);
+
+            Assert.IsNotNull(controller);
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Api.Tests/Controller/GiftControllerTests.cs]: In general you don't need to test the result of the constructor to make sure it is non-null. For this to be null would require a bug in the compiler. Simply omitting the assertion with a comment that you are merely checking to make sure the constructor did not through an exception is sufficient.
